### DISCUSSION TITLE
the way the kill-switch works

### DISCUSSION THREAD

**Discussion by ajay-gh** [gnuton/asuswrt-merlin.ng#686] [Ideas]
In earlier version of Merlin, the kill-switch worked exactly as it does not in the latest gnuton release. You turn it on, the vpn disconnected, and nothing gets through. The problem with it ? when you configure several vpn profiles (clients) and want to use one of them, you needed to make sure that all kill-switches in all other profiles were off. Later Merlin adopted a a more subtle strategy were any kill-switch is only working when that profile is active. When you turn that client off, and another client on, the connection works even with the kill-switch set in all other client. Could we consider this change in a coming gnuton release ?

**Comment by gnuton** [comment]
sure, can you spot the commit?

**Comment by ajay-gh** [comment]
I suppose you mean the Merlin commit (before and / or after) the modification ??
The Merlin change log (386.xx) seems to show that this change came about in version    386.3 (23-July-2021)
But I would not know how to find the corresponding commit number.
